### DISCUSSION TITLE
OLED SSH 1106 refresh rate  3 ~ 4sec

### DISCUSSION THREAD

**Discussion by niknak2020** [arendst/Tasmota#20893] [Q&A] [answered]
Issue OLED slow refresh rate with custom build (TasmoCompiler v11.2.1) 
I would like to see refresh rate of the display around 1 sec (time in seconds are counting flawless)

[Tasmota 13.3.0(TasmoCompiler-esp32generic) by Theo Arends](https://bit.ly/tasmota)

Sensors in use BME280, BH1750 and MH-Z19b

Hardware ESP8266 ESP32 Tested, 

Problem:
With original->  Tasmota Display Build OLED 1106 and DHT22 AM2302  I got a good refresh rate about 1sec 

Rule1 on AM2301#Temperature do DisplayText [s0y5x2] [T]/[s0x58y5] [tS] [f1x1y38]Temp. [f0s2x1y49]%value%~f8C endon on AM2301#Humidity do DisplayText [f1x75y38]Hum.[f0s2x75y49]%value% % endon

With Custom build I only get a slow refresh rate ... about 3-4 seconds 
I tried to use a ESP32 and different sensor setups ... tried to change the display refresh rate to 1 
tried with -tele and logging to 1 sec 

Any ideas what I could change?
<img width="557" alt="tasmota32" src="https://github.com/arendst/Tasmota/assets/65064968/d67b7a05-a995-4dc9-a6fc-861231cb66fe">

**Comment by sfromis** [comment]
This is not about the display, but about the sensor and driver. Can't say much about the backstory, but since the end of 2021, the sensor polling interval was extended to 4 seconds "solving some misreadings". Maybe you have an older build than that when you are getting data at a faster rate?

**Comment by niknak2020** [comment]
I used the newest release [Tasmota 13.3.0] for my custom online compiled ESP and the original "Display" build by Tasmota. 
Both devices were flashed via https://tasmota.github.io/install/ (web installer). 


Result: original build by Tasmota only supporting  the display and AM2301 --> fast refresh rate of the OLED display with my rule contain the command  [tS]. 

ESP32, ESP8266
Result with my custom build [Tasmota 13.3.0] via (TasmoCompiler v11.2.1) supporting all kinds of sensors I intent to use (BME280, BH1750 and MH-Z19b) resulting in a display refresh rate of 3-4 seconds (for the command [tS]). I also tested the command Displaymode 1 --> in this setting the clock counts just fine with a stable refresh rate for this setting. But when I use the tS command in a rule i only get about 3 sec refresh rate. 


Thanks for your help

**Reply by sfromis** [reply]
Again, what I mention is not about a "display refresh rate", but about when you can issue DisplayText commands. Rules based on AM2301 triggers are not available faster than a 4-second sensor poll rate.

Whatever you can do with `DisplayMode 1' is unrelated, as it is about the sensor values, not about the display.

**Reply by niknak2020** [reply]
Can I write a rule without a sensor like rule1 o time|seconds [tS] [z] for example? 

With the original Display build released by Tasmota my rule works perfect... that's so strange

**Reply by sfromis** [reply, marked as answer]
If what you want is a clock updated continuously, you should really not be mixing that into the sensor-driven rules. If you instead, once and for all, issue a command like this, Tasmota should keep the clock updated. Based on the [docs example](https://tasmota.github.io/docs/Displays/#display-json-variables); details should obviously be matched to your preferences.
`DisplayText [dv0:0:0:0:1:1:1:11:1:1:[tS]::]`
This could be done boot time, maybe from a `display.bat` file (in this case, omit the `DisplayText` command name).

This is also a good way to keep sensor values updated, without messing around with rules, like no rules at all.
